app/src/main/java/com/mpieterse/stride/ui/layout/central/views/NotificationsScreen.kt

```
@@ -1,37 +1,120 @@
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
 import com.mpieterse.stride.R
+import com.mpieterse.stride.core.permissions.NotificationPermissionManager
 import com.mpieterse.stride.ui.layout.central.components.CreateNotificationDialog
 import com.mpieterse.stride.ui.layout.central.components.NotificationItem
 import com.mpieterse.stride.ui.layout.central.models.NotificationData
 import com.mpieterse.stride.ui.layout.central.models.NotificationSettings
 import com.mpieterse.stride.ui.layout.central.viewmodels.NotificationsViewModel
+import kotlinx.coroutines.delay
 
 @Composable
 fun NotificationsScreen(
```

detekt (1.23.8)

[warning] The function NotificationsScreen appears to be too complex based on Cognitive Complexity (complexity: 26). Defined complexity threshold for methods is set to '25'

(detekt.complexity.CognitiveComplexMethod)